src/amo/components/AddonReviewCard/index.js

```diff
@@ -278,40 +281,55 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
       _config,
       addon,
       className,
+      clientApp,
       deletingReview,
       editingReview,
       errorHandler,
       flaggable,
       i18n,
-      siteUserHasReplyPerm,
+      lang,
       replyingToReview,
       review,
       shortByLine,
       showRating,
       siteUser,
+      siteUserHasReplyPerm,
       verticalButtons,
     } = this.props;
 
     let byLine;
+    const noAuthor = shortByLine || this.isReply();
 
     if (review) {
       const timestamp = i18n.moment(review.created).fromNow();
-      if (shortByLine || this.isReply()) {
-        // translators: Example in English: "posted last week"
-        byLine = i18n.sprintf(i18n.gettext('posted %(timestamp)s'), {
-          timestamp,
-        });
-      } else {
-        byLine = (
-          <span className="AddonReviewCard-authorByLine">
-            {/* translators: Example in English: "by UserName123, last week" */}
-            {i18n.sprintf(i18n.gettext('by %(authorName)s, %(timestamp)s'), {
+      const linkStart = oneLineTrim`<a href="/${lang}/${clientApp}/addon/${
+        review.addonSlug
+      }/reviews/${review.id}/">`;
+      const byLineString = noAuthor
+        ? // translators: Example in English: "posted last week"
+          i18n.gettext('posted %(linkStart)s%(timestamp)s%(linkEnd)s')
```

Comment:
Is there any benefit to injecting the HTML tags with sprintf? It makes the string more complicated for localizers. How about this?

```diff
diff --git a/src/amo/components/AddonReviewCard/index.js b/src/amo/components/AddonReviewCard/index.js
index abd386830..aba4842ae 100644
--- a/src/amo/components/AddonReviewCard/index.js
+++ b/src/amo/components/AddonReviewCard/index.js
@@ -301,17 +301,18 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
     const noAuthor = shortByLine || this.isReply();
 
     if (review) {
-      const timestamp = i18n.moment(review.created).fromNow();
-      const linkStart = oneLineTrim`<a href="/${lang}/${clientApp}/addon/${
-        review.addonSlug
-      }/reviews/${review.id}/">`;
+      const timestamp = `
+        <a href="/${lang}/${clientApp}/addon/${review.addonSlug}/reviews/${
+        review.id
+      }/">
+          ${i18n.moment(review.created).fromNow()}
+        </a>
+      `;
       const byLineString = noAuthor
         ? // translators: Example in English: "posted last week"
-          i18n.gettext('posted %(linkStart)s%(timestamp)s%(linkEnd)s')
+          i18n.gettext('posted %(timestamp)s')
         : // translators: Example in English: "by UserName123, last week"
-          i18n.gettext(
-            'by %(authorName)s, %(linkStart)s%(timestamp)s%(linkEnd)s',
-          );
+          i18n.gettext('by %(authorName)s, %(timestamp)s');
 
       byLine = (
         <span
@@ -322,8 +323,6 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
           dangerouslySetInnerHTML={sanitizeHTML(
             i18n.sprintf(byLineString, {
               authorName: review.userName,
-              linkStart,
-              linkEnd: '</a>',
               timestamp,
             }),
             ['a'],

```

Reply:
Duh. That's so much better. Thanks!